source/crud/tests/write/replaceOne-upsert.yml

```diff
@@ -1,48 +0,0 @@
-data:
-    - {_id: 1, x: 11}
-    - {_id: 2, x: 22}
-    - {_id: 3, x: 33}
-minServerVersion: '2.6'
-# See SERVER-5289 for why the collection data is only checked for server versions >= 2.6
```

Comment:
Splitting off `replaceOne-pre_2.6.yml` allowed us to relocate the upsert tests to the base `replaceOne.yml` file and remove this entirely.